src/Tizen.NUI/src/public/Common/Color.cs

```diff
@@ -953,31 +953,90 @@ public Color(float[] array) : this(Interop.Vector4.NewVector4(ValueCheck(array))
         /// <exception cref="ArgumentNullException">This exception is thrown when hexColor is null.</exception>
         /// <since_tizen> 6 </since_tizen>
         /// This will be public opened in tizen_6.0 after ACR done. Before ACR, need to be hidden as inhouse API.
         [EditorBrowsable(EditorBrowsableState.Never)]
-        public Color(string hexColor) : this(Interop.Vector4.NewVector4(), true)
+        public Color(string textColor) : this(Interop.Vector4.NewVector4(), true)
         {
             try
             {
-                if (null == hexColor)
+                if (null == textColor)
                 {
-                    throw new ArgumentNullException(nameof(hexColor));
+                    throw new ArgumentNullException(nameof(textColor));
                 }
-                hexColor = hexColor.Replace("#", "");
 
-                R = ((float)Convert.ToInt32(hexColor.Substring(0, 2), 16)) / 255.0f;
-                G = ((float)Convert.ToInt32(hexColor.Substring(2, 2), 16)) / 255.0f;
-                B = ((float)Convert.ToInt32(hexColor.Substring(4, 2), 16)) / 255.0f;
-                A = hexColor.Length > 6 ? ((float)Convert.ToInt32(hexColor.Substring(6, 2), 16)) / 255.0f : 1.0f;
+                textColor = textColor.ToLower();
+                textColor = textColor.Replace(" ", "");
+
+                if (textColor.Length>0 && textColor[0]=='#')
+                {
+                    textColor = textColor.Replace("#", "");
+                    int textColorLength = textColor.Length;
+
+                    if (textColorLength == 6 || textColorLength == 8) /* #RRGGBB or #RRGGBBAA */
+                    {
+                        R = ((float)Convert.ToInt32(textColor.Substring(0, 2), 16)) / 255.0f;
+                        G = ((float)Convert.ToInt32(textColor.Substring(2, 2), 16)) / 255.0f;
+                        B = ((float)Convert.ToInt32(textColor.Substring(4, 2), 16)) / 255.0f;
+                        A = textColor.Length > 6 ? ((float)Convert.ToInt32(textColor.Substring(6, 2), 16)) / 255.0f : 1.0f;
+                    }
+                    else if (textColorLength == 3 || textColorLength == 4) /* #RGB */
+                    {
+                        R = ((float)Convert.ToInt32(textColor.Substring(0, 1), 16)) / 15.0f;
+                        G = ((float)Convert.ToInt32(textColor.Substring(1, 1), 16)) / 15.0f;
+                        B = ((float)Convert.ToInt32(textColor.Substring(2, 1), 16)) / 15.0f;
+                        A = textColor.Length > 3 ? ((float)Convert.ToInt32(textColor.Substring(3, 1), 16)) / 15.0f : 1.0f;
+                    }
+                    else
+                    {
+                        throw new global::System.ArgumentException("Please check your color text code");
+                    }
+                }
+                else // example rgb(255,255,255) or rgb(255,255,255,1.0)
```

Comment:
I think If textColor.Length is 0, it can enter this else state.

Reply:
Yes it can, but we will throw exception that there is wrong string passed